File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/composite/player/PlayerGroup.java
package com.bigboxcode.designpattern.composite.player;

import java.util.ArrayList;
import java.util.List;

public class PlayerGroup implements Player {
    private List<Player> playerList = new ArrayList<>();

    @Override
    public void printDetails() {
        for (Player player: playerList) {
            player.printDetails();
        }
    }

    public void addElement(Player transport) {
        playerList.add(transport);
    }

    public void removeElement(Player transport) {
        playerList.remove(transport);
    }
}

File: design-patterns/ts/composite/player-group/player-group.ts
// player-group.ts

import Player from "./player";

class PlayerGroup implements Player {
    private playerList: Player[] = [];

    printDetails(): void {
        for (let player of this.playerList) {
            player.printDetails();
        }
    }

    addElement(transport: Player): void {
        this.playerList.push(transport);
    }

    removeElement(transport: Player): void {
        this.playerList.splice(this.playerList.indexOf(transport), 1);
    }
}

export default PlayerGroup;